fem/libceed/ceed.cpp

```diff
@@ -35,6 +35,9 @@ extern Ceed ceed;
 
 std::string ceed_path;
 
+extern std::unordered_map<std::string, CeedBasis> ceed_basis_table;
+extern std::unordered_map<std::string, CeedElemRestriction> ceed_restr_table;
```

Comment:
I think these should be in the `hpp` file, that's why you need to include `unordered_map` in device. It's strange to put a declaration in a `cpp` file.

Reply:
Not completely sure of what I'm saying though.